reciprocalspaceship/algorithms/intensity.py

```diff
@@ -0,0 +1,96 @@
+import warnings
+
+import numpy as np
+
+import reciprocalspaceship as rs
+
+
+def compute_intensity_from_structurefactor(
+    ds,
+    F_key,
+    SigF_key,
+    output_columns=None,
+    dropna=True,
+    inplace=False,
+):
+    """
+    Back-calculate intensities and approximate intensity error estimates from
+    structure factor amplitudes and error estimates
+
+    Intensity computed as I = SigF*SigF + F*F. Intensity error estimate
+    approximated as SigI = abs(2*F*SigF)
```

Comment:
I don't love the expression "back-calculate" -- it implies a workflow that isn't quite necessary. I would just say something like "Compute intensities (I) and uncertainty estimates (SigI) from structure factor amplitudes (F) and their uncertainties (SigF) using error propagation."